include/lbann/layers/activations/softmax.hpp

```diff
@@ -34,411 +32,335 @@
 #include "lbann/io/file_io.hpp"
 #include "lbann/utils/random.hpp"
 #include "lbann/models/model.hpp"
-#if defined(LBANN_HAS_CUDA) && defined(LBANN_SOFTMAX_CUDA)
-#include "lbann/layers/activations/softmax_cuda.hpp"
-#endif
 #include <unistd.h>
 #include <string>
-#include <typeinfo>
-#include <typeindex>
 
 #include <cassert>
 
+// Output has minimum value to avoid denormalized floats
 #define LBANN_ENABLE_SOFTMAX_CUTOFF
 
 namespace lbann {
 
+#ifdef LBANN_HAS_CUDNN
```

Comment:
Moving function prototypes from softmax_cuda.hpp allowed me to delete that file.